src/core/fieldformatter/qgsrangefieldformatter.cpp

```diff
@@ -0,0 +1,84 @@
+/***************************************************************************
+  qgsrangefieldformatter.cpp - QgsRangeFieldFormatter
+
+ ---------------------
+ begin                : 01/02/2018
+ copyright            : (C) 2018 by Alessandro Pasotti
+ email                : elpaso at itopen dot it
+ ***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+
+#include <QLocale>
+
+#include "qgsrangefieldformatter.h"
+
+#include "qgssettings.h"
+#include "qgsfield.h"
+#include "qgsvectorlayer.h"
+
+
+QString QgsRangeFieldFormatter::id() const
+{
+  return QStringLiteral( "Range" );
+}
+
+QString QgsRangeFieldFormatter::representValue( QgsVectorLayer *layer, int fieldIndex, const QVariantMap &config, const QVariant &cache, const QVariant &value ) const
+{
+  Q_UNUSED( cache )
+  Q_UNUSED( config )
+
+  if ( value.isNull() )
+  {
+    return QgsApplication::nullRepresentation();
+  }
+
+  QString result;
+
+  const QgsField field = layer->fields().at( fieldIndex );
+
+  if ( field.type() == QVariant::Double &&
+       field.editorWidgetSetup().config().contains( QStringLiteral( "Precision" ) ) &&
+       value.isValid( ) )
+  {
+    bool ok;
+    double val( value.toDouble( &ok ) );
+    if ( ok )
+    {
+      int precision( field.editorWidgetSetup().config()[ QStringLiteral( "Precision" ) ].toInt( &ok ) );
```

Comment:
if precision is not set, local for double is not applied... If I read correctly

Reply:
Yes, but it's just a safety measure, in theory it should never happen because "Precision" is in the config and it should be always set (but if it's not we get a crash without this check).